code/modules/mob/living/carbon/human/species_types.dm

```diff
@@ -623,6 +623,11 @@
 						"<span class='userdanger'>You throw up on the floor!</span>")
 	..()
 
+/datum/species/fly/check_weakness(obj/item/weapon, mob/living/attacker)
```

Comment:
Check_weakness is designed explicitly for things like this.  There is no way to retroactively change the amount of damage done in afterattack.